src/main/java/com/example/demo/service/FileService.java

```
@@ -0,0 +1,108 @@
+
+            if (resource.exists() || resource.isReadable()) {
+                return resource;
+            } else {
+                throw new RuntimeException("File is not readable");
+            }
+        } catch (MalformedURLException e) {
+            throw new RuntimeException("Error: " + e.getMessage());
+        }
+    }
+
+    @Override
+    public boolean deleteFile(String filename) {
+        boolean isDelete = true;
+        Path file = root.resolve(filename);
+        try {
+            if (Files.deleteIfExists(file)) {
+                isDelete = true;
+            } else {
+                isDelete = false;
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return isDelete;
+    }
+
+    @Override
+    public boolean updateFile(String existFileName, String newFileName) {
+        Path file = root.resolve(existFileName);
+        Path newFile = root.resolve(newFileName);
+        boolean isUpdate = true;
+        if (getFile(existFileName).exists()) {
+            try {
+                Files.move(file, newFile, StandardCopyOption.REPLACE_EXISTING);
+            } catch (IOException e) {
+                e.printStackTrace();
+                isUpdate = false;
+            }
+        } else {
+            isUpdate = false;
+        }
+        return isUpdate;
+    }
+
+    @Override
+    public FileInformation getInfoAboutFile(String file) {
+        Resource resource = getFile(file);
+        FileInformation fileInformation1 = new FileInformation();
+        if (resource.exists()) {
+            try {
+                File file1 = resource.getFile();
+                fileInformation1.setPathFile(file1.getAbsolutePath());
+                fileInformation1.setNameFile(file1.getName());
+                fileInformation1.setSize(file1.length());
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        } else {
+            fileInformation1 = null;
```

Try to never return null, in this case you should throw an exception or log that the file is missing